pkg/api/v1beta3/types.go

```diff
@@ -934,7 +934,7 @@ type ServiceSpec struct {
 	Ports []ServicePort `json:"ports" description:"ports exposed by the service"`
 
 	// This service will route traffic to pods having labels matching this selector. If null, no endpoints will be automatically created. If empty, all pods will be selected.
-	Selector map[string]string `json:"selector" description:"label keys and values that must match in order to receive traffic for this service; if empty, all pods are selected, if not specified, endpoints must be manually specified"`
+	Selector string `json:"selector" description:"label keys and values that must match in order to receive traffic for this service; if empty, all pods are selected, if not specified, endpoints must be manually specified"`
```

Comment:
@smarterclayton @bgrant0607 
I've always found misleading the fact that a selector is represented as a map
```
selector { 
"phase": "production" 
}
```
while in the command line the selector is `-l phase=production`. Even worst the map representation is equals to `labels` metatadata.
Having an array of requirements is an improvement but I would prefer to have `string` eventually introducing the special keyword `all` or `*` in the selector query language. And I would make use of the same for the command like:
`kubectl delete pods -l 'all'` for example.